### ISSUE TITLE
`read_csv` fails if BOM is present

### ISSUE THREAD

**Issue by amyskov** [modin-project/modin#3265]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 20.04
- **Modin version**: b58663bd8f2c8802bb45d7bafa2cc65ae988f959
- **Python version**: 3.8.10
- **Code we can use to reproduce**:
```
import codecs
import pandas
import modin.pandas as pd

test_file = "test_file.csv"

data = """col1,col2,col3,col4,col5,col6
0,2000-01-01,0,00:00:00,DJSfUncxsL,3361.303564068564
"""

with open(test_file, mode='w', encoding="utf-16-le") as f:
    f.write(codecs.BOM_UTF16_LE.decode("utf-16-le"))
    f.write(data)

df_pandas = pandas.read_csv(test_file, encoding="utf-16-le")
print(df_pandas)
df_pd = pd.read_csv(test_file, encoding="utf-16-le")
```

### Describe the problem

*BOM is Byte Ordering Mark - optional special bytes placed in the beginning of the file used for encoding.

### Source code / logs

```
   col1        col2  col3      col4        col5         col6
0     0  2000-01-01     0  00:00:00  DJSfUncxsL  3361.303564
UserWarning: Ray execution environment not yet initialized. Initializing...
To remove this warning, run the following python code before doing dataframe operations:

    import ray
    ray.init()

Traceback (most recent call last):
  File "bom.py", line 59, in <module>
    df_pd = pd.read_csv(test_file, encoding="utf-16-le")
  File "/modin/modin/pandas/io.py", line 131, in read_csv
    return _read(**kwargs)
  File "/modin/modin/pandas/io.py", line 59, in _read
    pd_obj = FactoryDispatcher.read_csv(**kwargs)
  File "/modin/modin/data_management/factories/dispatcher.py", line 172, in read_csv
    return cls.__factory._read_csv(**kwargs)
  File "/modin/modin/data_management/factories/factories.py", line 206, in _read_csv
    return cls.io_cls.read_csv(**kwargs)
  File "/modin/modin/engines/base/io/file_dispatcher.py", line 67, in read
    query_compiler = cls._read(*args, **kwargs)
  File "/modin/modin/engines/base/io/text/csv_dispatcher.py", line 161, in _read
    new_query_compiler = cls._get_new_qc(
  File "/modin/modin/engines/base/io/text/csv_dispatcher.py", line 303, in _get_new_qc
    new_index, row_lengths = cls._define_index(index_ids, index_col_md, index_name)
  File "/modin/modin/engines/base/io/text/csv_dispatcher.py", line 253, in _define_index
    row_lengths = cls.materialize(index_ids)
  File "/modin/modin/engines/ray/task_wrapper.py", line 82, in materialize
    return ray.get(obj_id)
  File "/miniconda3/envs/modin/lib/python3.8/site-packages/ray/_private/client_mode_hook.py", line 62, in wrapper
    return func(*args, **kwargs)
  File "/miniconda3/envs/modin/lib/python3.8/site-packages/ray/worker.py", line 1495, in get
    raise value.as_instanceof_cause()
ray.exceptions.RayTaskError(UnicodeDecodeError): ray::deploy_ray_func() (pid=3076612, ip=10.241.129.69)
  File "python/ray/_raylet.pyx", line 501, in ray._raylet.execute_task
  File "/miniconda3/envs/modin/lib/python3.8/site-packages/ray/util/tracing/tracing_helper.py", line 330, in _function_with_tracing
    return function(*args, **kwargs)
  File "/modin/modin/engines/ray/task_wrapper.py", line 40, in deploy_ray_func
    return func(**args)
  File "/modin/modin/backends/pandas/parsers.py", line 226, in parse
    pandas_df = pandas.read_csv(BytesIO(to_read), **kwargs)
  File "/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/util/_decorators.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 586, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 482, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 811, in __init__
    self._engine = self._make_engine(self.engine)
  File "/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 1040, in _make_engine
    return mapping[engine](self.f, **self.options)  # type: ignore[call-arg]
  File "/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/io/parsers/c_parser_wrapper.py", line 69, in __init__
    self._reader = parsers.TextReader(self.handles.handle, **kwds)
  File "pandas/_libs/parsers.pyx", line 542, in pandas._libs.parsers.TextReader.__cinit__
  File "pandas/_libs/parsers.pyx", line 734, in pandas._libs.parsers.TextReader._get_header
  File "pandas/_libs/parsers.pyx", line 843, in pandas._libs.parsers.TextReader._tokenize_rows
  File "pandas/_libs/parsers.pyx", line 1917, in pandas._libs.parsers.raise_parser_error
UnicodeDecodeError: 'utf-16-le' codec can't decode byte 0x0a in position 0: truncated data
```

**Comment by YarShev** [comment]
@amyskov , does the issue persist after #3134 merged?

**Comment by amyskov** [comment]
@YarShev , yes this issue is still persists, https://github.com/modin-project/modin/pull/3134 doesn't fixed in order to make PR more atomic. This issue should be fixed in the separate PR.

**Comment by vnlitvinov** [comment]
Fixed in [0.11.3](https://github.com/modin-project/modin/releases/tag/0.11.3)
